Makefile

```diff
@@ -360,6 +360,7 @@ $(include_packages):
 			--depends shadow-utils \
 			--rpm-digest sha256 \
 			--rpm-posttrans scripts/rpm/post-install.sh \
+			--rpm-os ${GOOS} \
```

Comment:
We squash each PR so your git history will get lost. Can you do this as a separate PR please?
Also, is there a reason to not hard-code linux here? Or can you create a windows/macos rpm?

Reply:
In theory, Windows and Mac RPMs are possible, and RPM tooling exists for Mac, so you can make and deploy Mac RPMs. I don't know why you'd want to, but it's possible. I'm fine with hardcoding linux if that's preferred.

Reply:
https://github.com/influxdata/telegraf/pull/11904 created with the commit as is. Can easily change to hardcoded linux if preferred.